src/finalisers/esm.ts

```diff
@@ -110,17 +110,20 @@ function getImportBlock(dependencies: ChunkDependencies, _: string): string[] {
 	return importBlock;
 }
 
-function getExportBlock(exports: ChunkExports, _: string): string[] {
+function getExportBlock(exports: ChunkExports, _: string, varOrConst: string): string[] {
 	const exportBlock: string[] = [];
 	const exportDeclaration: string[] = [];
 	for (const specifier of exports) {
+		let local = specifier.local;
 		if (specifier.exported === 'default') {
-			exportBlock.push(`export default ${specifier.local};`);
+			exportBlock.push(`export default ${local};`);
 		} else {
+			if (specifier.property) {
+				exportBlock.push(`${varOrConst} ${specifier.property}${_}=${_}${local};`);
```

Comment:
Looking at the usage together with the fact that it is now deconflicted, maybe `property` is not such a good name at all. Maybe it would be clearer if `property` was renamed and switched places with `local`:
- `local` would be the name of the introduced local variable (this would fit with the fact that `local` already corresponds to an existing local variable for non-synthetic exports)
- `property` could become `expression` and would correspond to an expression that returns the value of the local variable

```js
if (specifier.expression) {
				exportBlock.push(`${varOrConst} ${specifier.local}${_}=${_}${specifier.expression};`);
}
```

This would also avoid the need to reassign `local`, which is always something that makes me feel uneasy as this can make the code harder to follow.